server/src/main/webapp/WEB-INF/rails/webpack/views/pages/compare/stages/stages_widget.tsx

```diff
@@ -24,11 +24,32 @@ const classnames = bind(styles);
 
 interface Attrs {
   stages: Stages;
-  pipelineCounter?: number;
   onClick?: (stage: Stage) => void;
 }
 
 export class StagesWidget extends MithrilViewComponent<Attrs> {
+
+  static stageStatusClass(status: string) {
+    if (!status) {
+      return styles.unknown;
+    }
+
+    switch (status.trim().toLowerCase()) {
+      case "building":
+        return styles.building;
+      case "failed":
+        return styles.failed;
+      case "failing":
+        return styles.failing;
+      case "cancelled":
+        return styles.cancelled;
+      case "passed":
+        return styles.passed;
+      default:
+        return styles.unknown;
+    }
+  }
```

Comment:
how about this?

```js
static stageStatusClass(status?: string = "unknown") {
    return styles[status.trim().toLowerCase()];
}
```

Secondly, if we want to make that status can yield only specific values, we can change the type of the `status` argument from string to the possible values as follows:

```js
static stageStatusClass(status?: "passed" | "failed" |  "unknown" = "unknown") {
    return styles[status.trim().toLowerCase()];
}
```